### ISSUE TITLE
Introduce file ignorance for Composer distributions of packages

### ISSUE THREAD

**Issue by johnbillion** [wp-cli/wp-cli#5070]
Each of the repos for WP-CLI commands contain their own `.distignore` file ([example](https://github.com/wp-cli/entity-command/blob/master/.distignore)).

When a package is sent to Composer's Packagist, this file isn't used, because Packagist uses the `Source code` ZIP that GitHub builds for each release, and that uses `git archive` which doesn't read `.distignore` and uses `.gitattributes` instead.

In order to ignore unwanted files in Composer distributions, a `.gitattributes` file should be introduced which contains a list of files to ignore with a `export-ignore` property ([example](https://github.com/johnbillion/query-monitor/blob/master/.gitattributes)).

What would be the best way to introduce this file to all command repos?

**Comment by schlessera** [comment]
If we can define a universal `.gitattributes` that works for all repositories, I can just run a script that generates a pull request on all of them.

If we need to consider differences between packages, then we should start with a checklist of repositories to make sure we keep track of what was or wasn't done, use a script to create an issue in each package and finally create PRs manually to solve said issues.

**Comment by johnbillion** [comment]
I went through all the commands, and this is what I've got:

```
* text=auto

.distignore                      export-ignore
.editorconfig                    export-ignore
.gitattributes                   export-ignore
.github                          export-ignore
.gitignore                       export-ignore
.mailmap                         export-ignore
.travis.yml                      export-ignore
behat.yml                        export-ignore
bin                              export-ignore
ci                               export-ignore
features                         export-ignore
phpcs.xml.dist                   export-ignore
phpunit.xml                      export-ignore
phpunit.xml.dist                 export-ignore
tests                            export-ignore
travis-append.yml                export-ignore
travis-before_install-append.yml export-ignore
travis-cache-append.yml          export-ignore
```

**Comment by schlessera** [comment]
1. I think the `wp-cli.yml` file can be added as well, it just provides a convenient way to run the command under development.
2. What about `phpcs.xml, is that not needed?
3. I think we can also add `CONTRIBUTING.md` as well.
4. What about `README.md`? Does it make sense to have that inside of a dependency package? Composer metadata already points to the repository...
5. What about the `LICENSE` file? Does the package all of a sudden become unlicensed if we don't include that in a release distribution?

**Comment by johnbillion** [comment]
Yep let's ignore `wp-cli.yml`. I didn't see a repo containing `phpcs.xml`. If there is one, then let's ignore that too.

I didn't ignore `CONTRIBUTING.md`, `README.md`, and `LICENSE` mostly because I thought it's useful information for someone to see when looking through the files, but happy to ignore any of those if you don't think they're needed.
